### DISCUSSION TITLE
Tables in CLOG

### DISCUSSION THREAD

**Discussion by aykaramba** [rabbibotton/clog#335] [General]
Rabbi, is the following mean to be used instead of tables:

(let ((row (create-web-row main :padding t)))
      (create-web-container row :content "Grid Container 1" :column-size :third :class "w3-border w3-red")
      (create-web-container row :content "Grid Container 2" :column-size :third :class "w3-border w3-orange")
      (create-web-container row :content "Grid Container 3" :column-size :third :class "w3-border w3-blue"))

That is from Tutorial 24.  I looked at CLOG Builder generates for table date but I couldn't make heads of tails of it.  I am looking to present "tabular data" such as a list of of items and their properties.  Is that the proper component to use or does CLOG have a tables component that can receive w3css styling such as w3-striped?

**Comment by aykaramba** [comment]
The answer is we can programmatically create a table using (create-table ..) and (create-tr), etc.

**Comment by aykaramba** [comment]
Here are the html tags that we can create:

(create-a ...)
(create-p ...)
(create-br ...)
(create-h4 ...)
(create-div ...)
(create-img ...)
(create-form ...)
(create-span ...)
(create-term ...)
(create-audio ...)
(create-child ...)
(create-label ...)
(create-meter ...)
(create-panel ...)
(create-table ...)
(create-video ...)
(create-button ...)
(create-canvas ...)
(create-dialog ...)
(create-jquery ...)

**Comment by rabbibotton** [comment]
It is a ton more than that, plus you can custom create with create-child or
create-element or create-form-element

**Comment by aykaramba** [comment]
For any noobs that want to see how a table is constructed, here is a sample from clog test files:

```
                    (let* ((tbl (create-table win))
                           (cap (create-table-caption tbl :content "My Table"))
                           (rw))
                      (declare (ignore cap))
                      (set-border tbl :thin :solid :black)
                      (dotimes (y 10)
                        (setf rw (create-table-row tbl))
                        (dotimes (x 6)
                          (create-table-column rw :content (format nil "~A X ~A" x y)))))
```

**Comment by rabbibotton** [comment]
Appreciate btw having these posts it helps to know how easy the framework
is or is not and what would help documentation and video wise. Thanks!

People posting their issues especially in trying to understand are the most
valuable helpers to build CLOG

**Comment by aykaramba** [comment]
You are most welcome.  I am using CLOG to learn how to program (in general and program in Common Lisp).  Having real time feedback via CLOG is really making a difference for the learning process.  I hope to learn enough to help with ongoing documentation and other low barrier to entry stuff as I get better.  As well, one of the projects I want to develop is an app / service for creating and sharing code examples that nicely tie into the documentation system, but I have a long way to go before I can build something like that.

I can provide one more data point if you are interested:  There is NOTHING cooler than being able to program in one language and have it create the front end automagicaly.  If I wanted to learn programming using Laravel, I would be learning all of css, all of php, all of java script and the quirks of various browsers and other infrastructure at once.  What CLOG does is it makes web programming a reasonably sane and sensible process.

**Reply by kaeland** [reply]
As a Laravel developer, I really appreciate this comment and totally agree! In terms of your learning to program, please keep up the good work. As a fellow learner, it definitely takes time too. Learning to show myself grace each day. 

Thanks for the tip with creating tables as well!
